net/mwan3/files/etc/hotplug.d/iface/15-mwan3

```diff
@@ -9,24 +9,29 @@
 [ -n "$INTERFACE" ] || exit 2
 
 if [ "$ACTION" == "ifup" ]; then
-        [ -n "$DEVICE" ] || exit 3
+	[ -n "$DEVICE" ] || exit 3
 fi
 
+mwan3_lock "$ACTION" "$INTERFACE"
 config_load mwan3
 config_get_bool enabled globals 'enabled' '0'
-[ ${enabled} -gt 0 ] || exit 0
+[ "${enabled}" -gt 0 ] || {
+	mwan3_unlock "$ACTION" "$INTERFACE"
+	exit 0
+}
 
-mwan3_lock "$ACTION" "$INTERFACE"
 mwan3_init
 mwan3_set_connected_iptables
 mwan3_set_custom_ipset
-mwan3_unlock "$ACTION" "$INTERFACE"
 
 config_get enabled $INTERFACE enabled 0
```

Comment:
Can this be "config_get_bool" here to be consistent with the global enabled check?